pandas/core/indexes/base.py

```diff
@@ -2935,32 +2935,42 @@ def _union(self, other, sort):
         lvals = self._values
         rvals = other._values
 
-        if sort is None and self.is_monotonic and other.is_monotonic:
+        if (
```

Comment:
can you add a comment here on when these branches are taken (similar to what you did below).

Reply:
Done